code/ATMOSPHERICS/components/binary_devices/passive_gate.dm

```diff
@@ -184,7 +184,15 @@
 	SSnanoui.update_uis(src)
 
 /obj/machinery/atmospherics/binary/passive_gate/attackby(obj/item/W, mob/user, params)
-	if(!istype(W, /obj/item/wrench))
+	if(istype(W, /obj/item/pen))
+		var/t = copytext(stripped_input(user, "Enter the name for the gate.", "Rename", name), 1, MAX_NAME_LEN)

```

Comment:
`copytext` is not required here due to `stripped_input` already capping the character limit